### ISSUE TITLE
create a custom markup for inline blocks

### ISSUE THREAD

**Issue by shcheklein** [iterative/blog#16]
We need to write an HTML snippet to embed a link.

Can we utilize mdx or create a custom MD markup to do this?

**Comment by iAdramelk** [comment]
Depending on the syntax that we want, there is multiple ways to achieve it:

1. Use [remark-custom-blocks](https://github.com/zestedesavoir/zmarkdown/tree/master/packages/remark-custom-blocks) to add simple syntax blocks.
2. Use pseudo-html tags with [react-rehype](https://using-remark.gatsbyjs.org/custom-components/)
3. Use actual MDX.

Can you write examples of the snippets that we need to create with all fields that these snippets have?

**Comment by iAdramelk** [comment]
Or we can just transform markdown AST in any way we want.

**Comment by shcheklein** [comment]
Yep, you can check it here (or almost in any other blog post):

https://github.com/iterative/blog/edit/master/content/blog/2019-12-14-november-19-dvc-heartbeat.md

```html
<a href="https://www.deploymachinelearning.com/" class="external-link-preview">
  <section class="elp-content-holder">
    <div class="elp-description-holder">
      <h4 class="elp-title">Deploy Machine Learning Models with Django</h4>
      <div class="elp-description">Version 1.0 (04/11/2019) Piotr Płoński The demand for Machine Learning (ML) applications is growing. Many resources…</div>
      <div class="elp-link">deploymachinelearning.com</div>
    </div>
    <div class="elp-image-holder">
      <img src="/uploads/images/2019-12-14/deploy-machine-learning-models.png" />
    </div>
  </section>
</a>
```

**Comment by shcheklein** [comment]
@iAdramelk considering your three options, could you convert that example above into different representations so that we can pick the simplest one?

**Comment by iAdramelk** [comment]
We have four fields in the snippet above:

- *href*: "https://www.deploymachinelearning.com/"
- *title*: "Deploy Machine Learning Models with Django"
- *description*: "Version 1.0 (04/11/2019) Piotr Płoński The demand for Machine Learning (ML) applications is growing. Many resources..."
- *picture*: "/uploads/images/2019-12-14/deploy-machine-learning-models.png"

This number of fields excludes custom blocks, but other two options remains.

We can do with that something like:

```
<external-link
  href="https://www.deploymachinelearning.com/"
  title="Deploy Machine Learning Models with Django"
  description="Version 1.0 (04/11/2019) Piotr Płoński The demand for Machine Learning (ML) applications is growing. Many resources..."
  picture="/uploads/images/2019-12-14/deploy-machine-learning-models.png"
></external-link>
```

Or:

```
<external-link
  href="https://www.deploymachinelearning.com/" 
  picture="/uploads/images/2019-12-14/deploy-machine-learning-models.png"
>
  <external-link-title>Deploy Machine Learning Models with Django"</external-link-title>
  <external-link-description>Version 1.0 (04/11/2019) Piotr Płoński The demand for Machine Learning (ML) applications is growing. Many resources...</external-link-description>
</external-link>
```

Or with mdx:

```
<ExternalLink
  href="https://www.deploymachinelearning.com/"
  title="Deploy Machine Learning Models with Django"
  description="Version 1.0 (04/11/2019) Piotr Płoński The demand for Machine Learning (ML) applications is growing. Many resources..."
  picture="/uploads/images/2019-12-14/deploy-machine-learning-models.png"
/>
```

**Comment by shcheklein** [comment]
@iAdramelk all three look reasonable to me. The first one probably is the easiest one?

**Comment by iAdramelk** [comment]
Yes, the thirst one is probably the easiest. To add `mdx` we would need to replace our markdown rendering plugin for `gatsby` (which is easy), but the `mdx` plugin didn't have out of the box plugins for the things like syntax highlighting, auto-resizing images in markdown articles, gist embedding, etc. so we will need to re-add them manually.

**Comment by sekmet** [comment]
Hello, I got this done.
Please check the updated DVC blog live here: 
https://inspiring-brown-042ba4.netlify.com/

I’d love to hear from you guys!
Best regards
Carlos Guimaraes

**Comment by shcheklein** [comment]
Fixed by #65
